recipes/harfbuzz/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://harfbuzz.org"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "patches/*.patch"]
+    generators = "cmake"
+    short_paths = True
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_freetype": [True, False],
+        "with_icu": [True, False],
+        "with_glib": [True, False],
+        "with_gdi": [True, False],
+        "with_uniscribe": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_freetype": True,
+        "with_icu": False,
+        "with_glib": False,
+        "with_gdi": True,
+        "with_uniscribe": True
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    _cmake = None
+
+    def requirements(self):
+        if self.options.with_freetype:
+            self.requires("freetype/2.10.1")
+        if self.options.with_icu:
+            self.requires("icu/64.2")
+        if self.options.with_glib:
+            self.requires("glib/2.65.0")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        else:
+            del self.options.with_gdi
+            del self.options.with_uniscribe
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake_compiler_flags(self, cmake):
+        flags = []
+        compiler = str(self.settings.compiler)
+        if compiler in ("clang", "apple-clang"):
+            flags.append("-Wno-deprecated-declarations")
+        cmake.definitions["CMAKE_C_FLAGS"] = " ".join(flags)
+        cmake.definitions["CMAKE_CXX_FLAGS"] = cmake.definitions["CMAKE_C_FLAGS"]
+
+        return cmake
+
+    def _configure_cmake_macos(self, cmake):
+        if tools.is_apple_os(self.settings.os):
+            cmake.definitions["CMAKE_MACOSX_RPATH"] = True
+        return cmake
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake = self._configure_cmake_compiler_flags(self._cmake)
+        self._cmake = self._configure_cmake_macos(self._cmake)
+        self._cmake.definitions["HB_HAVE_FREETYPE"] = self.options.with_freetype
+        self._cmake.definitions["HB_BUILD_TESTS"] = False
+        self._cmake.definitions["HB_HAVE_ICU"] = self.options.with_icu
+        self._cmake.definitions["HB_HAVE_GLIB"] = self.options.with_glib
+
+        if self.options.with_icu:
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = "17"
```

Comment:
Given this C++ standard requirement with ICU, we should add a check in the `configure()` method to validate if the provided compiler support C++17.

Reply:
I just tested on my machine: `conan create /mnt/c/pf/conan-center-index/recipes/harfbuzz/all/ harfbuzz/2.6.8@ -o harfbuzz:with_icu=True -s compiler.cppstd=11 -b missing` passes without error
the cmake definition was originally added in this commit https://github.com/bincrafters/conan-harfbuzz/pull/10/commits/4ebf77bb7a11892e5595f5c638cf54ee725a9fdb#diff-2b1d42f71f22b7ea0412d7602dec166fR87 but it looks it was not necessary